api/v1beta1/common.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/endpoint"
+)
+
+// GlanceAPITemplate defines the desired state of GlanceAPI
+type GlanceAPITemplate struct {
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=glance
+	// ServiceUser - optional username used for this service to register in glance
+	ServiceUser string `json:"serviceUser"`
```

Comment:
I'm wondering if we need both Glance.Spec.ServiceUser and Glance.Spec.GlanceAPI[Internal|External].ServiceUser as well in the top level. Or we can simply copy the Glance.Spec.ServiceUser to the GlanceAPI.Spec.ServiceUser directly?

Reply:
Yes, we actually do in the controller, which means we don't need this in the template.